docs/wskdeploy_apigateway_open_api_spec.md

```diff
@@ -0,0 +1,177 @@
+
+It shows how to:
+- update the Action named ‘hello_world’ to expose it to the gateway.
+- specify the API's endpoint that will trigger the action via an Open API Specification.
+
+### Manifest file
+#### _Example: “Hello world” action with project config to point to the API spec_
+```yaml
+project:
+  config: open_api_spec.json
+  packages:
+    hello_world_package:
+      version: 1.0
+      license: Apache-2.0
+      actions:
+        hello_world:
+          function: src/hello.js
+          web-export: true
+```
+### Open API Specification
+#### _Example: "Hello world" open api specification_
+```json
+{
+    "swagger": "2.0",
+    "info": {
+        "version": "1.0",
+        "title": "Hello World API"
+    },
+    "basePath": "/hello",
+    "schemes": [
+        "https"
+    ],
+    "consumes": [
+        "application/json"
+    ],
+    "produces": [
+        "application/json"
+    ],
+    "paths": {
+        "/world": {
+            "get": {
+                "description": "Returns a greeting to the user!",
+                "operationId": "hello_world",
+                "responses": {
+                    "200": {
+                        "description": "Returns the greeting.",
+                        "schema": {
+                            "type": "string"
+                        }
+                    }
+                }
+            }
+        }
+
+    },
+    "x-gateway-configuration": {
+    "assembly": {
+      "execute": [
+        {
+          "operation-switch": {
+            "case": [
+              {
+                "operations": [
+                  "getHello"
+                ],
+                "execute": [
+                  {
+                    "invoke": {
+                      "target-url": "https://openwhisk.ng.bluemix.net/api/some/action/path.http",
+                      "verb": "keep"
+                    }
+                  }
+                ]
+              }
+            ],
+            "otherwise": []
+          }
+        }
+      ]
+    }
+  }
+}
+```
+*NOTE*: Some providers such as IBM may have changed these keys. For example, IBM uses `x-ibm-configuration` instead of `x-gateway-configuration`.
+
+There are two major differences from _"Hello World" API_ example:
+- the root key is now project as the open api specification is a project wide concept.
+- a new `config` key specifying where the Open API Specification is located.
+
+The `config` key under `project` in the manifest file specifies where the Open API Specification is located. The keyword `config` was chosen to remain consistent with the OpenWhisk CLI flag option. The Open API Specification describes in a JSON document the the base path, endpoint, HTTP verb, and other details describing the API. For example, the document above describes a GET endpoint at `/hello/world` that recieves JSON as input and returns JSON as output.
```

Comment:
is it `config` or `config-file` in CLI? it says `config-file` here https://github.com/apache/openwhisk-cli/blob/07993fd22c5618ee153eade186369d7d6084d1b7/commands/api.go#L174